src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -16,154 +20,182 @@ namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
+        public static readonly DependencyProperty CollectionProperty =
+            DependencyProperty.Register(nameof(Collection), typeof(ITrackCollectionViewModel), typeof(ZuneTrackCollection), new PropertyMetadata(null, (d, e) => ((ZuneTrackCollection)d).OnCollectionChanged((ITrackCollectionViewModel?)e.OldValue, (ITrackCollectionViewModel?)e.NewValue)));
 
         /// <summary>
-        /// Holds the <see cref="ZuneTrackCollection"/> listview.
+        /// The collection to display.
         /// </summary>
-        public ListView? PART_Selector { get; private set; }
+        public ITrackCollectionViewModel? Collection
+        {
+            get => (ITrackCollectionViewModel?)GetValue(CollectionProperty);
+            set => SetValue(CollectionProperty, value);
+        }
 
         /// <summary>
-        /// Creates a new instace for <see cref="ZuneTrackCollection"/>.
+        /// The track items being displayed in the UI. Contains additional functionality specific to this view.
         /// </summary>
-        public ZuneTrackCollection()
+        public ReadOnlyObservableCollection<ZuneTrackCollectionItem> TrackItems { get; }
+
+        /// <summary>
+        /// If true, there is only 1 distinct track artist for all tracks in the provided <see cref="Collection"/>.
+        /// </summary>
+        public bool AllTrackArtistsAreTheSame => TrackItems.Any(x => x.ShouldShowArtistList);
+
+        /// <inheritdoc/>
+        protected override async Task LoadMore()
         {
+            if (Collection is null)
+                return;
+
+            if (!Collection.PopulateMoreTracksCommand.IsRunning)
+                await Collection.PopulateMoreTracksCommand.ExecuteAsync(25);
         }
 
         /// <inheritdoc/>
-        protected override void OnApplyTemplate()
+        protected override void CheckAndToggleEmpty()
         {
-            base.OnApplyTemplate();
+            if (Collection is null)
+                return;
 
-            PART_ArtistColumn = GetTemplateChild(nameof(PART_ArtistColumn)) as TextBlock;
-            PART_Selector = GetTemplateChild(nameof(PART_Selector)) as ListView;
+            if (!Collection.PopulateMoreTracksCommand.IsRunning && Collection.TotalTrackCount == 0)
+                SetEmptyVisibility(Visibility.Visible);
+        }
 
-            Guard.IsNotNull(PART_ArtistColumn, nameof(PART_ArtistColumn));
+        private void OnCollectionChanged(ITrackCollectionViewModel? oldValue, ITrackCollectionViewModel? newValue)
+        {
+            foreach (var item in TrackItems)
+                DetachEvents(item);
 
-            if (PART_Selector != null)
-                PART_Selector.SelectionChanged += PART_Selector_SelectionChanged;
+            _trackItems.Clear();
+            OnPropertyChanged(nameof(AllTrackArtistsAreTheSame));
 
-            PART_ArtistColumn.Visibility = Visibility.Collapsed;
-        }
+            _ = Execute();
```

Comment:
Hmmm, if there are not instructions to execute afterwards, don't discard Async methods.

Reply:
Changed. This is the only place where we can change the containing method to be async, the rest of them we don't get a choice.